spec/models/ambassador_task_assignment_spec.rb

```diff
@@ -5,48 +5,52 @@
   let(:ambassador) { FactoryBot.create(:ambassador) }
   let(:ambassador_task) { FactoryBot.create(:ambassador_task) }
 
-  describe "pending_complete" do
-    let!(:ambassador_task_assignment1) { FactoryBot.create(:ambassador_task_assignment, user: ambassador, ambassador_task: ambassador_task) }
-    let!(:ambassador_task_assignment2) { FactoryBot.create(:ambassador_task_assignment, user: ambassador, completed_at: Time.now - 1.day) }
-    it "is complete" do
-      ambassador.reload
-      expect(ambassador.ambassador_task_assignments.completed).to eq([ambassador_task_assignment2])
-      expect(ambassador.ambassador_task_assignments.locked_completed.pluck(:id)).to eq([ambassador_task_assignment2.id])
-      expect(ambassador.ambassador_task_assignments.pending_completion.pluck(:id)).to eq([ambassador_task_assignment1.id])
-      ambassador_task_assignment1.update_attributes(completed_at: Time.now - 1.minute)
-      ambassador.reload
-      expect(ambassador.ambassador_task_assignments.pending_completion.pluck(:id)).to eq([ambassador_task_assignment1.id])
+  describe "#completed" do
```

Comment:
nit: I know this isn't TDD doctrine, but for tests like this where you're adding a bunch of objects in the database, in the interest of preventing the test suite from slowing down, I lean towards trying to consolidate and only create the objects once - unless -

- Creating them multiple times makes the tests easier to understand
- The different objects don't overlap much, so there would be little (or no) time savings
- Too many things are tested, so when modifying or adding to anything in the app it breaks the fat tests
- ... or anything else tilts the balance at all

I'd **much** prefer comprehensive, readable tests to incomplete tests. This is mainly a justification for why I wrote just one test block originally.